tests/test_noninteractive_mode.py

```diff
@@ -0,0 +1,96 @@
+""" Non-interactive tests. """
+import subprocess
+from mssqltestutils import (
+    create_mssql_cli,
+    # create_test_db,
+    # clean_up_test_db,
+)
+
+# TODO: split up into multiple classes and create automated way to generate query files
+class TestNonInteractive:
+    """ Tests non-interactive features. """
+
+    file_root = './tests/test_query_results/'
+
+    # @classmethod
+    # def setup_class(cls):
+    #     cls.test_db = create_test_db()
+
+    # @classmethod
+    # def teardown_class(cls):
+    #     clean_up_test_db(cls.test_db)
+
+
+    # TESTS
+    def test_session_closure_query_valid(self):
+        """ Test session closure for valid query. """
+        assert self.is_processed_closed("select 1")
+
+    def test_session_closure_query_invalid(self):
+        """ Test session closure for invalid query. """
+        assert self.is_processed_closed("asdfasoifjas")
+
+    def test_query_large(self):
+        """ Test against query with over 1000 lines. """
+        query_str = "SELECT * FROM STRING_SPLIT(REPLICATE(CAST('X,' AS VARCHAR(MAX)), 1024), ',')"
+        file_baseline = self.file_root + 'big.txt'
```

Comment:
A neat convention would be to name the baseline file the same as the test name.

Reply:
I actually decided to use the parametrize pattern in pytest instead, which I validated prints useful error messages in the event of a failure.